src/library/scala/collection/Map.scala

```diff
@@ -29,25 +29,14 @@ trait Map[K, +V]
 
   def canEqual(that: Any): Boolean = true
 
-  override def equals(o: Any): Boolean = o match {
-    case that: Map[K, _] =>
-      (this eq that) ||
-      (that canEqual this) &&
-      (this.size == that.size) && {
-        try {
-          val checker = new AbstractFunction1[(K, V),Boolean] {
-            override def apply(kv: (K,V)): Boolean = {
-              that.getOrElse(kv._1, Map.DefaultSentinel) == kv._2
-            }
-          }
-          this forall checker
-        } catch {
-          case _: ClassCastException => false
-        }
-      }
-    case _ =>
-      false
-  }
+  override def equals(o: Any): Boolean =
+    (this eq o.asInstanceOf[AnyRef]) || (o match {
+      case map: Map[K, _] if map.canEqual(this) =>
```

Comment:
should this be `MapOps[_, _, _]`? as it is now, for all `m: Map[_, _]`, `m != m.view`

(I realise this PR doesn't change the behaviour, but it does seem worth changing to me)

the same comment can be made about the other collections (except `Set`, which doesn't have its own `View` type, which raises other questions)

Reply:
That’s an interesting point. I think we had debated about the equality on views and concluded that they would not be comparable to other collections. I don’t remember where the discussion is, though…

Reply:
I searched a bit and found the following.

At https://github.com/scala/collection-strawman/pull/436#issuecomment-364964726

> We also had meeting discussions about views and equality. We couldn’t find a sensible definition of equality for views. Especially because equality between collections is not defined at the level of iterable but only in Seq, Set and Map. A practical consequence of this is that views can not extend Seq, Set or Map.

At https://github.com/scala/collection-strawman/issues/13#issuecomment-285384395

> we define equality at the level of Seq and Set. View is yet another basic collection type which would never be equal to a Seq or Set.

https://github.com/scala/bug/issues/10201#issuecomment-383663116: SeqView is not reflexive

> Fixed in 2.13.x for 2.13.0-M4. Views are compared by identity in the new collection library.